vibi-dpu/src/github/prs.rs

```diff
@@ -128,3 +129,75 @@ pub async fn get_and_store_pr_info(repo_owner: &str, repo_name: &str, access_tok
         log::error!("[get_and_store_pr_info] No PR info available for PR number: {:?} repository: {:?} repo_owner{:?}", pr_number, repo_name, repo_owner);
     }
 }
+
+pub async fn pr_reviewer_handles(repo_owner: &str, repo_name: &str,
+        pr_number: &str, pr_head_commit: &str,access_token: &str)
+        -> Option<Vec<String>> {
+    let response_opt = all_pr_reviews(access_token,
+        repo_owner, repo_name, pr_number).await;
+    if response_opt.is_none() {
+        log::error!("[pr_reviewer_handles] Unable to get reviewer handles from gh api");
+        return None;
+    }
+    let response = response_opt.expect("Uncaught empty pr reviewers response");
+    let parse_result = response.json::<Vec<Value>>().await;
+    if parse_result.is_err() {
+		let e = parse_result.expect_err("No error in parsing");
+		log::error!(
+			"[pr_reviewer_handles] Failed to parse JSON: {:?}",
+			e
+		);
+		return None;
+	}
+	let reviewer_list_result = parse_result.expect("Uncaught error in parsing reviewers list data");
+    // Initialize a vector to store reviewer handles
+    let mut reviewer_handles = Vec::new();
+
+    // Process the review list
+    for review in reviewer_list_result {
+        let state = review["state"].as_str().unwrap_or_default();
+        let commit_id = review["commit_id"].as_str().unwrap_or_default();
+        if state == "APPROVED" && commit_id == pr_head_commit {
+            // Extract reviewer login
+            if let Some(login) = review["user"]["login"].as_str() {
+                reviewer_handles.push(login.to_string());
+            }
+        }
+    }
+    Some(reviewer_handles)
+}
+
+async fn all_pr_reviews(access_token: &str,
```

Comment:
This function name is misleading. Can we have something like: `get_all_reviewers_for_gh_pr` since right now it only gets the data for gh.

Reply:
This one is only called inside prs.rs, so this name might be safe. The one called in approval.rs is `pr_reviewer_handles`, which I am planning to call like - `github::prs::pr_reviewer_handles`, I have also done similar stuff in add_comment etc